src/control/controlobjectscript.cpp

```diff
@@ -127,13 +127,27 @@ void ControlObjectScript::disconnectAllConnectionsToFunction(const QJSValue& fun
     }
 }
 
+bool ControlObjectScript::isCallbackExecuting() const {
+    return m_callbackExecuting;
+}
+
+void ControlObjectScript::setCallbackExecuting(bool executing) {
+    m_callbackExecuting = executing;
+}
+
 void ControlObjectScript::slotValueChanged(double value, QObject*) {
     // Make a local copy of m_connectedScriptFunctions first.
     // This allows a script to disconnect a callback from inside the
-    // the callback. Otherwise the this may crash since the disconnect call
-    // happens during conn.function.call() in the middle of the loop below.
-    const QVector<ScriptConnection> connections = m_scriptConnections;
+     const QVector<ScriptConnection> connections = m_scriptConnections;
+    // Flag to inhibit triggering connections during callback execution
+
     for (auto&& conn: connections) {
-        conn.executeCallback(value);
+        if (!this->m_callbackExecuting) {
```

Comment:
```suggestion
        if (!m_callbackExecuting) {
```
this-> prefix is redundant